vscode/activation.rb

```diff
@@ -1,3 +1,3 @@
-env = ENV.filter_map { |k, v| "#{k}RUBY_LSP_VS#{v}" if v.dup.force_encoding(Encoding::UTF_8).valid_encoding? }
+env = ENV.filter_map { |k, v| v = v.dup.force_encoding(Encoding::UTF_8); "#{k}RUBY_LSP_VS#{v}" if v.valid_encoding? }
```

Comment:
I guess we can simplify this and just try to encode straight away and rescue the conversion error.

```suggestion
env = ENV.filter_map do |k, v|
 "#{k}RUBY_LSP_VS#{v.encode(Encoding::UTF_8)}"
rescue Encoding::UndefinedConversionError
  nil
end
```

Reply:
Interesting. I didn't expect `encode` to raise even when it's possible to force the encoding to UTF-8. You're right, let's go with the original then.

```ruby
env = ENV.filter_map do |k, v|
  utf_8_value = v.dup.force_encoding(Encoding::UTF_8)
  "#{k}RUBY_LSP_VS#{utf_8_value}" if utf_8_value.valid_encoding?
end
```